src/main/java/spoon/support/reflect/declaration/InvisibleArrayConstructor.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (C) 2006-2017 INRIA and contributors
+ * Spoon - http://spoon.gforge.inria.fr/
+ *
+ * This software is governed by the CeCILL-C License under French law and
+ * abiding by the rules of distribution of free software. You can use, modify
+ * and/or redistribute the software under the terms of the CeCILL-C license as
+ * circulated by CEA, CNRS and INRIA at http://www.cecill.info.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the CeCILL-C License for more details.
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL-C license and that you accept its terms.
+ */
+package spoon.support.reflect.declaration;
+
+/**
+ * This class is used to represent the constructor of an array when calling with an expression like
+ * <pre>
+ *     String[]::new
+ * </pre>
+ */
+public class InvisibleArrayConstructor<T> extends CtConstructorImpl<T> {
```

Comment:
By convention, should end by Impl: `InvisibleArrayConstructorImpl`, and should be in `spoon.support.reflect.declaration` (all implementation classes are in spoon.support)

Reply:
It is in spoon.support. I will rename it.